test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -14002,6 +13989,42 @@ def add_mom_dyn_res(self, custom_resource, script_body=None,
         self.add_config(a)
         return res_file
 
+    def enable_cgroup_cset(self):
+        """
+        Configure and enable cgroups hook
+        """
+        # check if cgroups subsystems including cpusets are mounted
+        file = os.path.join(os.sep, 'proc', 'mounts')
+        mounts = self.du.cat(self.hostname, file)['out']
+        pat = 'cgroup /sys/fs/cgroup'
+        if str(mounts).count(pat) >= 6 and str(mounts).count('cpuset') >= 2:
+            pbs_conf_val = self.du.parse_pbs_config(self.hostname)
+            f1 = os.path.join(pbs_conf_val['PBS_EXEC'], 'lib',
+                              'python', 'altair', 'pbs_hooks',
+                              'pbs_cgroups.CF')
```

Comment:
does this file exist on an execution-only host?

Reply:
This file does not exist on the execution host.  Since we are running PTL on the server node, PTL is able to pick up that file.  The server then pushes out the cgroups config to the moms.  I attached tests that I ran on uv-01 and td-07 (e.g. TestVerifyLogOutput_uv-01_td-07.txt) that show that only the cpuset machine got vnoded and the non-cpuset machine that has sockets did not get vnoded.

Reply:
Ok, looks good